docker/android-system.sh

```diff
@@ -116,49 +559,264 @@ EOF
+}
 
-    # list from https://elinux.org/Android_toolbox
-    for tool in cat chmod chown cmp cp ctrlaltdel date df dmesg du \
-        hd id ifconfig iftop insmod ioctl ionice kill ln log ls \
-        lsmod lsof lsusb md5 mkdir mount mv nandread netstat notify \
-        printenv ps reboot renice rm rmdir rmmod route schedtop sendevent \
-        setconsole setprop sleep smd start stop sync top touch umount \
-        uptime vmstat watchprops wipe; do
-        ln -s /system/bin/toolbox "/system/bin/${tool}"
-    done
+android_repo_v11() {
+    sync art
+    sync bionic
+    sync bootable/recovery
+    sync build/blueprint
+    sync build/make
+    sync build/soong
+    sync external/clang
+    sync external/compiler-rt
+    sync external/elfutils
+    sync external/fmtlib
+    sync external/golang-protobuf
+    sync external/gwp_asan
+    sync external/jemalloc
+    sync external/jemalloc_new
+    sync external/libcxx
+    sync external/libcxxabi
+    sync external/libunwind
+    sync external/libunwind_llvm
+    sync external/llvm
+    sync external/lzma
+    sync external/mksh
+    sync external/scudo
+    sync external/zlib
+    sync prebuilts/build-tools
+    sync prebuilts/clang-tools
+    sync prebuilts/clang/host/linux-x86
+    sync prebuilts/gcc/linux-x86/host/x86_64-linux-glibc2.17-4.8
+    sync prebuilts/go/linux-x86
+    sync prebuilts/sdk
+    sync prebuilts/vndk/v28
+    sync prebuilts/vndk/v29
+    sync system/core
+    sync system/sepolicy
 
-    echo "127.0.0.1 localhost" > /system/etc/hosts
+    case "${ARCH}" in
+        arm)
+            sync external/arm-optimized-routines
+            sync prebuilts/gcc/linux-x86/arm/arm-linux-androideabi-4.9
+        ;;
+        arm64)
+            sync external/arm-optimized-routines
+            sync prebuilts/gcc/linux-x86/arm/arm-linux-androideabi-4.9
+            sync prebuilts/gcc/linux-x86/aarch64/aarch64-linux-android-4.9
+        ;;
+        x86)
+            sync prebuilts/gcc/linux-x86/x86/x86_64-linux-android-4.9
+        ;;
+        x86_64)
+            sync prebuilts/gcc/linux-x86/x86/x86_64-linux-android-4.9
+        ;;
+    esac
 
-    purge_packages
+    # avoid build tests
+    rm bionic/tests/Android.mk
+    rm bionic/tests/Android.bp
+    rm bionic/benchmarks/Android.bp
+    rm bionic/tests/libs/Android.bp
+    rm bionic/tests/headers/Android.bp
+    rm bionic/tests/headers/posix/Android.bp
 
-    popd
+    # make sure we don't build benchmarks or apex
+    rm -r bionic/apex
+    rm -r bionic/benchmarks/
 
-    rm -rf "${td}"
-    rm "${0}"
+    # libziparchive has tests in the header, remove them
+    local libziparchive_h="system/core/libziparchive/include/ziparchive/zip_writer.h"
+    sed -i -e 's/#include <gtest\/gtest_prod.h>//g' "${libziparchive_h}"
+    sed -i -e 's/FRIEND_TEST(zipwriter, WriteToUnseekableFile);//g' "${libziparchive_h}"
+
+    remove_tests
+}
+
+android_repo_v12() {
+    # FIXME: this is a work in progress, and is unlikely to ever
```

Comment:
I've kept this in since it's a work-in-progress. It might never finish, but it's a good starting point if anyone can ever do this work in the future.